dgraphtest/graphql.go

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2023 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dgraphtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"runtime/debug"
+	"strings"
+	"time"
+
+	"github.com/dgraph-io/dgraph/dql"
+	"github.com/dgraph-io/dgraph/x"
+)
+
+type GqlSchema struct {
+	Id              string
+	Schema          string
+	GeneratedSchema string
+}
+
+type ProbeGraphQLResp struct {
+	Healthy             bool `json:"-"`
+	Status              string
+	SchemaUpdateCounter uint64
+}
+
+type GraphQLResponse struct {
+	Data       json.RawMessage        `json:"data,omitempty"`
+	Errors     x.GqlErrorList         `json:"errors,omitempty"`
+	Extensions map[string]interface{} `json:"extensions,omitempty"`
+}
+
+var retryableUpdateGQLSchemaErrors = []string{
+	"errIndexingInProgress",
+	"is already running",
+	"retry again, server is not ready", // given by Dgraph while applying the snapshot
+	"Unavailable: Server not ready",    // given by GraphQL layer, during init on admin server
+}
+
+func RequireNoGQLErrors(resp *GraphQLResponse) {
+	if resp != nil {
+		panic(fmt.Sprintf("Expected value not to be nil."))
+	}
+	RequireNoError(resp.Errors)
+}
+
+func RequireNoError(err error) {
+	if err != nil {
+		debug.PrintStack()
+		panic(fmt.Sprintf("required no GraphQL errors, but received: %s\n", err.Error()))
+	}
+}
+
+func Equal(expected interface{}, actual interface{}) bool {
+	// Use the reflect package to get the values of the interfaces
+	value1 := reflect.ValueOf(expected)
+	value2 := reflect.ValueOf(actual)
+
+	// If the types are different, they are not equal
+	if value1.Type() != value2.Type() {
+		return false
+	}
+
+	// Compare the values of the interfaces
+	return reflect.DeepEqual(expected, actual)
+}
+
+func (hc *HTTPClient) SafelyUpdateGQLSchema(schema string, alphaPort string) *GqlSchema {
```

Comment:
We already have UpdateGQLSchema function that is already available in dgraphtest package?

Reply:
Resolved

Reply:
I still see this function here